ark/utils/plot_utils.py

```diff
@@ -85,26 +84,84 @@ def tif_overlay_preprocess(segmentation_labels, plotting_tif):
     return formatted_tif
 
 
-def create_overlay(segmentation_labels, plotting_tif, alternate_segmentation=None):
+def create_overlay(fov, segmentation_dir, data_dir,
+                   img_overlay_chans, seg_overlay_comp, alternate_segmentation=None):
     """Take in labeled contour data, along with optional mibi tif and second contour,
     and overlay them for comparison"
-
     Generates the outline(s) of the mask(s) as well as intensity from plotting tif. Predicted
     contours are colored red, while alternate contours are colored white.
 
     Args:
-        segmentation_labels (numpy.ndarray):
-            2D numpy array of labeled cell objects
-        plotting_tif (numpy.ndarray):
-            2D or 3D numpy array of imaging signal
+        fov (str):
+            The name of the fov to overlay
+        segmentation_dir (str):
+            The path to the directory containing the segmentatation data
+        data_dir (str):
+            The path to the directory containing the nuclear and whole cell image data
+        img_overlay_chans (list):
+            List of channels the user will overlay
+        seg_overlay_comp (str):
+            The segmentted compartment the user will overlay
         alternate_segmentation (numpy.ndarray):
             2D numpy array of labeled cell objects
-
     Returns:
         numpy.ndarray:
             The image with the channel overlay
     """
 
+    # load the specified fov data in
+    plotting_tif = load_utils.load_imgs_from_dir(
+        data_dir=data_dir,
+        files=[fov + '.tif'],
+        xr_dim_name='channels',
+        xr_channel_names=['nuclear_channel', 'membrane_channel']
+    )
+
+    # verify that the provided image channels exist in plotting_tif
```

Comment:
I don't understand why this doesn't fail. We name the channels 'nuclear_channel' and 'membrane_channel', so if the user passes a channel name this isn't going to match. But in the notebook_utils_test this passes when you provide ['chan0', 'chan1'] as an input

Reply:
The reason it didn't fail was because I never actually injected the call into the notebook. When I did, it failed spectacularly. Fixed.